### PR TITLE
http-gateway/ui: opentelemetry

### PR DESCRIPTION
**Pull request by PatrikMatiasko**

### CODE REVIEW

File: http-gateway/web/src/common/services/opentelemetry.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { context, trace } from '@opentelemetry/api'
+
+let webTracer = undefined
+
+const init = (appName = '') => {
+  const provider = new WebTracerProvider()
+  provider.addSpanProcessor(new SimpleSpanProcessor(new ConsoleSpanExporter()))
+  provider.register({
+    contextManager: new ZoneContextManager(),
+    propagator: new B3Propagator(),
```
Comment (jkralik) [inline http-gateway/web/src/common/services/opentelemetry.js]:
```
const api = require("@opentelemetry/api");
const { W3CTraceContextPropagator } = require("@opentelemetry/core");

...
propagator: new W3CTraceContextPropagator(),
```

File: http-gateway/web/package.json [hunk 2]
```diff
@@ -6,6 +6,14 @@
     "@auth0/auth0-react": "^1.8.0",
     "@babel/cli": "^7.16.0",
     "@craco/craco": "^6.4.3",
+    "@opentelemetry/api": "^1.1.0",
+    "@opentelemetry/context-zone": "^1.3.1",
+    "@opentelemetry/instrumentation": "^0.29.2",
+    "@opentelemetry/instrumentation-fetch": "^0.29.2",
+    "@opentelemetry/instrumentation-xml-http-request": "^0.29.2",
+    "@opentelemetry/propagator-b3": "^1.3.1",
```
Comment (jkralik) [inline http-gateway/web/package.json]: This propagator is not currently supported by http-server - you can see it in header as `b3`:
image


Pls, could you replace it with core W3C propagator which is implemented in core ? https://npm.io/package/@opentelemetry/core

File: http-gateway/web/src/common/services/opentelemetry.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    instrumentations: [
+      new FetchInstrumentation({
+        ignoreUrls: [/localhost:3000\/sockjs-node/],
+        clearTimingResources: true,
+      }),
+      new XMLHttpRequestInstrumentation({
+        ignoreUrls: [/localhost:3000\/sockjs-node/],
+      }),
+    ],
+  })
```
Comment (jkralik) [inline http-gateway/web/src/common/services/opentelemetry.js:33]: Do we need this lines ?

File: http-gateway/web/src/containers/devices/rest.js [hunk 4]
```diff
@@ -65,16 +70,19 @@ export const getDevicesResourcesApi = ({
...
+    () =>
+      fetchApi(
+        `${security.getGeneralConfig().httpGatewayAddress}${
+          devicesApiEndpoints.DEVICES
+        }/${deviceId}/resources${href}?timeToLive=${ttl}&${interfaceGetParam(
+          currentInterface
+        )}`,
+        { method: 'PUT', body: data, timeToLive: ttl }
+      ),
+    'update-resource-for-device'
```
Comment (jkralik) [inline http-gateway/web/src/containers/devices/rest.js]: update-device-resource

File: http-gateway/web/src/containers/devices/rest.js [hunk 5]
```diff
@@ -51,10 +52,14 @@ export const getDevicesResourcesApi = ({
...
-      devicesApiEndpoints.DEVICES
-    }/${deviceId}/resources${href}?${interfaceGetParam(currentInterface)}`
+  withTelemetry(
+    () =>
+      fetchApi(
+        `${security.getGeneralConfig().httpGatewayAddress}${
+          devicesApiEndpoints.DEVICES
+        }/${deviceId}/resources${href}?${interfaceGetParam(currentInterface)}`
+      ),
+    'get-resource-for-device'
```
Comment (jkralik) [inline http-gateway/web/src/containers/devices/rest.js]: get-device-resource

**Review by jkralik — APPROVED** [review]

**Comment by jkralik** [comment]
Goob job 👍

**Review by ondrejtomcik — APPROVED** [review]

**Comment by ondrejtomcik** [comment]
@jkralik can you please release it tmrw?

**Pull request merged**